cmd/helm/repo.go

```diff
@@ -47,15 +47,22 @@ func runRepoAdd(cmd *cobra.Command, args []string) error {
 	}
 	name, url := args[0], args[1]
 
+	if err := addRepository(name, url); err != nil {
+		return err
+	}
+	fmt.Println(name + " has been added to your repositories")
+	return nil
+}
+
+func addRepository(name, url string) error {
 	if err := downloadIndexFile(name, url); err != nil {
-		return errors.New("Oops! Looks like " + url + " is not a valid chart repository or cannot be reached\n")
+		return errors.New("Looks like " + url + " is not a valid chart repository or cannot be reached\n")
```

Comment:
I am referring to `err` returned from `downloadIndexFile()`.  It is checked for `nil` but not added to `errors.New(...)`